### PR TITLE
added command to duplicate line

### PR DESCRIPTION
**Pull request by thewhitetulip**
Expected working:
1. if there is single/multiple cursor(s) without selection then prepend the entire line for each cursor
2. if there is/are single/multiple cursor(s) with selection then only copy the selected text and prepend it before the selection

### CODE REVIEW

File: line.go [hunk 1]
```diff
@@ -42,6 +50,34 @@ type (
...
+			lline := v.Line(r.End())
+			t := v.Substr(lline)
+			at := t + "\n"
+			v.Insert(e, lline.Begin(), at)
+		} else {
+			// if selection isn't empty then we want to copy
+			// and prepend the copied text.
+			selectedText := v.Substr(r)
+			v.Insert(e, r.Begin(), selectedText)
+		}
```
Comment (erbridge) [inline line.go:73]: A fair amount of this is repeated. How about:

``` go
r := sel.Get(i)
suffix := ""

if r.Empty() {
    r = v.Line(r.End())
    suffix = "\n"
}

t := v.Substr(r) + suffix
v.Insert(e, r.Begin(), t)
```

File: line.go [hunk 2]
```diff
@@ -12,6 +12,13 @@ import (
 )
 
 type (
+	// Duplicate will duplicate the line
+	// If no selection, then duplicate the line the cursor is on
+	// otherwise, duplicate the last line of the selection.
```
Comment (erbridge) [inline line.go]: "Duplicate copies all of the current selections, inserting each copy before the original. If any of the selections are empty, instead, the entire line containing that cursor is copied and inserted before the original."

File: line.go [hunk 3]
```diff
@@ -42,6 +50,34 @@ type (
 	}
 )
 
+// Run executes the DuplicateLine command
```
Comment (erbridge) [inline line.go]: Missing `.` :).

**Comment by thewhitetulip** [comment]
@erbridge Travis does not look happy.

**Comment by thewhitetulip** [comment]
@erbridge Changes done and travis is happy :D

**Pull request merged**